### PR TITLE
[C++ worker] add node-ip-address params to C++ worker

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?
- No node-ip-address param now. A hard code of "127.0.0.1" will bring some problems in multiple nodes tests.

### CODE REVIEW

File: cpp/src/ray/util/process_helper.cc [hunk 1]
```diff
@@ -96,9 +96,12 @@ void ProcessHelper::RayStart(CoreWorkerOptions::TaskExecutionCallback callback)
   options.enable_logging = true;
   options.log_dir = log_dir;
   options.install_failure_signal_handler = true;
-  options.node_ip_address = "127.0.0.1";
+  std::string node_ip = ConfigInternal::Instance().node_ip_address.empty()
+                          ? "127.0.0.1"
+                          : ConfigInternal::Instance().node_ip_address;
+  options.node_ip_address = node_ip;
```
Comment (kfstorm) [inline cpp/src/ray/util/process_helper.cc:102]:
```suggestion
  options.node_ip_address = ConfigInternal::Instance().node_ip_address;
  if (options.node_ip_address.empty()) {
    options.node_ip_address = "127.0.0.1";
  }
```

File: cpp/src/ray/worker/default_worker.cc [hunk 2]
```diff
@@ -10,7 +10,7 @@ using namespace ::ray::api;
 
 int default_worker_main(int argc, char **argv) {
   RAY_LOG(INFO) << "CPP default worker started";
-  RAY_CHECK(argc == 8);
+  RAY_CHECK(argc == 9);
```
Comment (kfstorm) [inline cpp/src/ray/worker/default_worker.cc:13]: Why not use gflags to parse arguments?

**Comment by kfstorm** [comment]
Is it possible to cover this by UT?

File: cpp/src/ray/util/process_helper.cc (hunk 1 above)
Comment (SongGuyang) [reply cpp/src/ray/util/process_helper.cc:102]: Why?

File: cpp/src/ray/worker/default_worker.cc (hunk 2 above)
Comment (SongGuyang) [reply cpp/src/ray/worker/default_worker.cc:13]: I can make another PR to use gflags.

**Comment by SongGuyang** [comment]
@kfstorm No way to make a test using two ip addresses.

File: cpp/src/ray/util/process_helper.cc (hunk 1 above)
Comment (kfstorm) [reply cpp/src/ray/util/process_helper.cc:102]: In this way you don't need to create a local variable and repeat `ConfigInternal::Instance().node_ip_address` twice. Just a nit.

Comment (SongGuyang) [reply cpp/src/ray/util/process_helper.cc:102]: We need assign this value twice.
`options.node_ip_address = node_ip;` and `options.raylet_ip_address = node_ip;`

**Review by raulchen — APPROVED** [review]

**Pull request merged**